### ISSUE TITLE
PTSD Accidental Deletion of whole backup (Request --read-only)

### ISSUE THREAD

**Issue by g0kuz** [kaczmarkiewiczp/rcloneExplorer#224]
Can you implement **--read-only** on your android app. I have chills just thinking of how easy you can delete your whole library with a few buttons.

Also to change this setting, a password would be required or even better, the encryption password.

**Comment by patrykcoding** [comment]
Not really sure if I see a good reason for this feature. 

- Rclone doesn't even ask for confirmation before deleting files, so I would argue that on desktop it's even easier to delete all your files.
- Many remotes support Trash, so if you delete something accidentally, you can restore it.
- A Google Drive remote can be created with read-only scope (not really helpful if you're using other providers)

Also, `read only` suggests that it would not be possible to upload new files, or modify existing ones (rename, move). Is that what you have in mind?

**Comment by g0kuz** [comment]
Yes, because I use rclone android to only stream and download, there's no reason to have the delete feature as i edit and upload through the computer.

Maybe at the start of the app you could implement the --read-only througout the app, meaning only "clear data"  under app info can reset it.

**Comment by patrykcoding** [comment]
Adding password protection on the delete function shouldn't be hard to do, so I can implement that.

Right now I took a bit of a break from this project, only fixing bugs and updating rclone version. I'm hoping to start working on it again within a month. I still want to add password protection when opening the app, so once I have that functionality done, I'll add password protection on the delete function.

**Comment by g0kuz** [comment]
Yea, but adding password protection on deletion doesn't prevent any changes like renaming/moving files.

Best case, if you ever have time, is to implement a --read-only. That's the only thing that makes me safe without any worry my files are touched.
